clevertap-core/src/main/java/com/clevertap/android/sdk/utils/CTJsonConverter.java

```diff
@@ -90,6 +91,13 @@ public static JSONObject from(DeviceInfo deviceInfo, CoreMetaData coreMetaData,
             evtData.put("hgt", deviceInfo.getHeight());
             evtData.put("dpi", deviceInfo.getDPI());
             evtData.put("dt", DeviceInfo.getDeviceType(deviceInfo.getContext()));
+
+            // If locale is set by the client then use that, otherwise fetch it from the device
+            String locale = CleverTapAPI.getLocale();
```

Comment:
No need to add `CleverTapAPI` as a dependency in this class if you set the locale in DeviceInfo class. Refer to previous comment